Lib/locale.py

```diff
@@ -321,6 +327,11 @@ def delocalize(string):
         string = string.replace(dd, '.')
     return string
 
+def localize(string, grouping=False, monetary=False):
+    """Parses a string as locale number according to the locale settings."""
+    float(string)
```

Comment:
@cedk 

> Ensure that it is a string that represent a valid number.

Hmm, I'm not sure that I agree with this means of validating the strings. Primarily because it will result in a non-obvious error message for users. For example, `localize('')` would result in:

```
ValueError: could not convert string to float: ''
```

I suspect that this message would make zero sense to users unless they looked at the implementation of the function, which should not be a requirement for diagnosing error messages.

IMO, you should at the very least catch the `ValueError` and then re-raise it with a more informative error message. For example:

```suggestion
    try:
        float(string)
    except ValueError:
        raise ValueError(f"{string} is not a valid number.")
```

Reply:
Indeed I added this for https://github.com/python/cpython/pull/15275#pullrequestreview-287111359. But as delocalize does not perform any check and passing a non valid number will not fail, I think it is better to relax the check.

Reply:
Hmm, that review link seems to be longer working. Who was it that originally suggested that change? Particularly if it was from @matrixise (or other core dev), I would highly recommended further discussing the removal of that change with them.